html/includes/table/fdb-search.inc.php

```diff
@@ -106,8 +116,10 @@
         }
         if ($entry['port_id'] == $endpoint_portid) {
             $endpoint_img = "<i class='fa fa-star fa-lg' style='color:green' aria-hidden='true' title='This indicates the most likely endpoint switchport'></i>";
+            $dnsname = gethostbyaddr(reset(explode(',', $entry['ipv4_address']))) ?: 'N/A';
```

Comment:
Do you need to put this in a function which caches the response otherwise you might just be sending the same query if the IP is listed multiple times.

Reply:
Not sure I understand correctly, but if there are multiple adresses in `$entry['ipv4_address']` the function `reset` will make sure only one is used.

Reply:
Not sure if I understand correctly, but if there are multiple ip addresses in `$entry['ipv4_address']` the `reset` function will make sure we only use one of them (the first).

Reply:
The results in the query can have the same ip in multiple rows. I realise with reset you are getting just one but the next row might have the same IPs in and the lookup would be done again

Reply:
hm, the gethostbyaddr lookup is only performed if `$entry['port_id'] == $endpoint_portid`, and `$endpoint_portid` can only be one. So unless there are multiple rows with the same  `$entry['port_id']` in the query result I don't see how that would happen.

Reply:
True, ignore me then :)

Reply:
I'd definitely be interested to see someone use this with a good amount of results returned to understand the implications.

Reply:
well, I really appreciate your feedback! You have far better understanding of the code base so it's really valuable. And I would also like more testing of this in other environments...